classes/Authentication/SAML/XDSamlAuthentication.php

```
@@ -164,12 +164,12 @@ private function notifyAdminOfNewUser($user, $samlAttributes, $linked, $error =
         "Additional SAML Attributes ----------------------------------\n\n" .
         print_r($samlAttributes, true);
 
-        $mail->setBodyText($body);
+        $mail->Body = $body;
         try {
             $mail->send();
             return true;
         } catch (Exception $e) {
-            return $e->getMessage();
+            return $e->getMessage() . "\n" . $mail->ErrorInfo;
```

Rather than return `true` in one case and a string in another case, you should re-throw the exception and catch it in the calling routine. This makes error handling easier.